src/netml/pparser/parser.py

```diff
@@ -1041,11 +1051,11 @@ def _pcap2pandas(self):
 
         Returns:
           DataFrame with one packet per row.
-            column names are the keys from pcap_to_dict plus
+            column names are those of _pcap_columns plus
             'ip_dst_int', 'ip_src_int', 'mac_dst_int', 'mac_dst_int'
 
         """
-        self.df = pd.DataFrame(self._iter_pcap_dict())
+        self.df = pd.DataFrame(self._iter_pcap_dict(), columns=self._pcap_columns)
```

Comment:
This simply guards against columns missing from the `DataFrame` – and risking a `KeyError` below – in the edge-case that the stream is empty.

Pandas is instructed what columns to construct, rather than expecting the library to infer this from the stream.

Factoring the sequence of columns out is perhaps slightly less straightforward but a lot more DRY than the alternative.